django/db/models/expressions.py

```diff
@@ -245,6 +247,10 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
         ])
         return c
 
+    @property
+    def conditional(self):
+        return isinstance(self.output_field, fields.BooleanField)
```

Comment:
I wanted to add a note that I think this will enable usages such as `.filter(F('boolean_field'))` that we might want to either explicitly disable (through `F.conditional = False`) or test them (which will require some adjustments on Oracle).

Reply:
`F` doesn't inherit from `BaseExpression` so currently it doesn't have `conditional` property and cannot be used in `filter()` or `exclude()` we can change this in the future.